frame/staking/src/pallet/mod.rs

```diff
@@ -1281,11 +1303,15 @@ pub mod pallet {
 			#[pallet::compact] additional: u32,
 		) -> DispatchResult {
 			ensure_root(origin)?;
-			ValidatorCount::<T>::mutate(|n| *n += additional);
+			ValidatorCount::<T>::mutate(|n| {
+				*n = <T::ElectionProvider as ElectionProviderBase>::MaxWinners::get()
+					.min(*n + additional);
```

Comment:
I would err here.

Reply:
Unsafe math in the runtime?! This will panic on `Staking::increase_validator_count(RuntimeOrigin::root(), u32::MAX)`.  
It is permissioned, but I would rather not like to find out.